### PR TITLE
[Merged by Bors] - feat: port Geometry.Manifold.Algebra.LeftInvariantDerivation

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/Algebra/LeftInvariantDerivation.lean [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+theorem toFun_eq_coe : X.toFun = ⇑X :=
+  rfl
+#align left_invariant_derivation.to_fun_eq_coe LeftInvariantDerivation.toFun_eq_coe
+
+#noalign left_invariant_derivation.coe_to_linear_map
+
+@[simp]
+theorem toDerivation_eq_coe : X.toDerivation = X :=
+  rfl
+#align left_invariant_derivation.to_derivation_eq_coe LeftInvariantDerivation.toDerivation_eq_coe
```
Comment (urkud) [inline Mathlib/Geometry/Manifold/Algebra/LeftInvariantDerivation.lean:87]:
```suggestion
-- Porting note: now LHS is the same as RHS
#noalign left_invariant_derivation.to_derivation_eq_coe
```

File: Mathlib/Geometry/Manifold/Algebra/LeftInvariantDerivation.lean [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+variable (r)
+
+@[simp]
+theorem coe_smul : ⇑(r • X) = r • ⇑X :=
+  rfl
+#align left_invariant_derivation.coe_smul LeftInvariantDerivation.coe_smul
+
+@[simp]
+theorem lift_smul (k : 𝕜) : (↑(k • X) : Derivation 𝕜 C^∞⟮I, G; 𝕜⟯ C^∞⟮I, G; 𝕜⟯) = k • X :=
```
Comment (urkud) [inline Mathlib/Geometry/Manifold/Algebra/LeftInvariantDerivation.lean:216]:
```suggestion
theorem lift_smul (k : 𝕜) : (k • X).1 = k • X.1 :=
```

**Review by urkud — COMMENTED** [review]
Fixes for linter

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**